add homework6 (from Java Beginner Book [p.28-30] and Excel file [clas…

iryna-pantsialeyeva opened the pull request:
I'd be grateful if you review my PR.

iryna-pantsialeyeva committed:
add homework6 (from Java Beginner Book [p.28-30] and Excel file [class 4])

mrnchernova commented on src/task6/MaxRandomNumber.java:
```diff
@@ -0,0 +1,22 @@
+        Arrays.sort(arrayRandomNumber);
+        System.out.println(arrayRandomNumber[size-1]);
+    }
+}
```
Для наглядности лучше выводить максимальный элемент с новой строки

i-velichko commented on src/task6/MarkArray.java:
```diff
@@ -0,0 +1,56 @@
+
+        int maxMark = markArray[0];
+        int minMark = markArray[0];
+        int indexMaxMark = 0;
```
why did you create this variable?

iryna-pantsialeyeva replied:
Because I wanted to use it than I changed the algorithm. Nevertheless you are absolutely right - I should check highlighted fields to exclude the redundant code. Many thanks for your comment.

i-velichko commented on src/task6/MarkArray.java:
```diff
@@ -0,0 +1,56 @@
+        int maxMark = markArray[0];
+        int minMark = markArray[0];
+        int indexMaxMark = 0;
+        int indexMinMark = 0;
```
and this

iryna-pantsialeyeva replied:
See the explanation above

i-velichko commented on src/task6/MarkArray.java:
```diff
@@ -0,0 +1,56 @@
+package task6;
+/*Создать массив оценок произвольной длины, вывести максимальную и минимальную оценку, её (их) номера.*/
```
How about use array sorting in this task? Looks like we can simplify our code

iryna-pantsialeyeva replied:
I've just done. Please find the second way to resolve the task in my 2nd commit.
But as far as I understand the recommended aproach with method Array.sort makes code more complicated. 
We need to make a new object - new array and make an additional variable. Because we should not only to find max and min values but also we have to find their index or indexes (if the number of marks with max/min value is more than 1). 
Using method Array.sort elements of initial array changes their "place" in array and their indexes as well. After the implementation this method we can't get the right index(es). So we should create a duplicate of initial array.
If I was wrong I'd apreciate it if you could share with me the algorithm mentioned by you that can simplify code.
Many thanks in advance.

i-velichko replied:
But I don't see a condition specifying the requirement to specify array indexes.

i-velichko commented on src/task6/MaxRandomNumber.java:
```diff
@@ -0,0 +1,22 @@
+        }
+
+        Arrays.sort(arrayRandomNumber);
+        System.out.println(arrayRandomNumber[size-1]);
```
Formatting (Ctrl + Alt + L)  everywhere and add message please. And good! Why you didn't use the same approach for another tasks?

iryna-pantsialeyeva replied:
Because I wittingly try to use different approaches. I'm just learning to write a code. The implementation of different approaches hepls me to memorize new info.

i-velichko commented on src/task6/SentenceFrequencyDictionary.java:
```diff
@@ -0,0 +1,64 @@
+
+    static String[] textToSentenceParser(String text) {
+        String[] sentence = text.split("[.!?]");
+        return sentence;
```
Can we exclude the variable declarations here and put that in a "return"?

iryna-pantsialeyeva replied:
Yes, we can.

i-velichko commented on src/task6/SentenceFrequencyDictionary.java:
```diff
@@ -0,0 +1,64 @@
+    static String[] sentenceToWordParser(String sentence) {
+
+        String[] word = sentence.trim().split(",\\s|\\s");
+        return word;
```
Can we exclude the variable declarations here and put that in a "return"?

iryna-pantsialeyeva replied:
It makes sense but I suppose it's just a matter of taste.

i-velichko replied:
No, it's not a matter of taste. It's an extra line of code that makes it less readable. You should try to think about the colleagues you will be working with who will need to read your code.

i-velichko commented on src/task6/SentenceFrequencyDictionary.java:
```diff
@@ -0,0 +1,64 @@
+public class SentenceFrequencyDictionary {
+    public static void main(String[] args) {
+
+        String text = "Я буду маліцца і сэрцам, і думамі, Распетаю буду маліцца душой. Каб чорныя долі з " +
```
За Мову жырны плюс)

i-velichko commented on src/task6/SentenceFrequencyDictionary.java:
```diff
@@ -0,0 +1,64 @@
+        }
+    }
+
+    static String[] textToSentenceParser(String text) {
```
why "static" ?)

iryna-pantsialeyeva replied:
Because of non-static methods can not be referenced from a static content (method "main" is static).

i-velichko replied:
good)

iryna-pantsialeyeva committed:
add update of MarkArray and SentenceFrequencyDictionary by request in PR
MarkArray: as requested Array.sort are used
SentenceFrequencyDictionary: 2 unused variables are deleted, variables declaration moved to return
